### PR TITLE
Add reducers

### PR DESCRIPTION
**Pull request by emmanuellmhango**
I have implemented the following requirements:

- Added Redux Toolkit.
- Structured the application files as used by [ducks pattern](https://github.com/erikras/ducks-modular-redux) for the Redux files.
- Added book's actions and reducer (in /src/redux/books/books.js) 
- Added categories actions and reducer (in /src/redux/categories/categories.js) 
- Configured the Redux Store (/src/redux/configureStore.js).

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -1,15 +1,13 @@
 const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (julie-ify) [inline src/redux/categories/categories.js]: - [ ] In keeping with the duck's pattern, kindly re-assign your action type here.

  ```
  const CHECK_STATUS = 'bookstore/categories/CHECK_STATUS'
  ```

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,27 +1,25 @@
 const ADD_BOOK = 'ADD_BOOK';
 const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (julie-ify) [inline src/redux/books/books.js]: - [ ] Let's implement the duck's pattern correctly.

  ```
  const ADD_BOOK = 'bookstore/books/ADD_BOOK'
  const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK'
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @emmanuellmhango,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 2 above)
Comment (emmanuellmhango) [reply src/redux/books/books.js]: Thank you @julie-ify , I have made the correction

File: src/redux/categories/categories.js (hunk 1 above)
Comment (emmanuellmhango) [reply src/redux/categories/categories.js]: Thank you @julie-ify , I have made the correction

**Review by ha-manel — APPROVED** [review]
# STATUS: APPROVED ✅

## Hi @emmanuellmhango 👋

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
**Congratulations! 🎉**



## Highlights:
* Well-written PR description ✔️ 
* All tests passed ✔️ 


------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**